### ISSUE TITLE
Support custom CV splitters (LeaveOneOut) + pooled scoring in AAPred.eval

### ISSUE THREAD

**Issue by breimanntools** [breimanntools/aaanalysis#397]
## Problem

`AAPred.eval` hardcodes `StratifiedKFold(n_splits=n_cv)` in the backend
(`aaanalysis/prediction/_backend/aa_pred/aa_pred_eval.py::_score_cv`) and its
frontend guard `check_n_cv` caps `n_cv` at the smallest class count. On a
small, imbalanced benchmark — e.g. the γ-secretase substrate set (63 substrates
vs. 14 known non-substrates) — this makes **LeaveOneOut** cross-validation
impossible through the shipped API, so users drop back to hand-rolled sklearn:

```python
balanced_accuracy_score(y, cross_val_predict(SVC(kernel="linear"), X, y, cv=LeaveOneOut()))
```

There is a second, subtler gap: even if `eval` accepted a splitter, `_score_cv`
uses `cross_val_score` (score **per fold, then average**). Under LeaveOneOut
each test fold has a single sample, so per-fold `balanced_accuracy` is
degenerate and its average does **not** equal the study's metric, which pools
every held-out prediction and scores **once**. Closing this gap therefore means
adding both a custom-CV argument *and* a pooled-prediction scoring principle.

The absence forces every small-n evaluation in a use case to reimplement CV by
hand, which is exactly the boilerplate the prediction layer was meant to absorb.

## Goal

Let `AAPred.eval` accept an arbitrary scikit-learn CV splitter (including
`LeaveOneOut`) and score it via **pooled out-of-fold predictions**, reproducing
`balanced_accuracy_score(y, cross_val_predict(est, X, y, cv=cv))` exactly — with
the default integer-`n_cv` path (StratifiedKFold, per-fold scoring) left
byte-identical.

## Requirements

- [ ] `AAPred.eval` accepts a CV **splitter object** in addition to the int
      `n_cv` (e.g. a new `cv=` parameter, or `n_cv` accepting a splitter);
      backward compatible — int keeps the current StratifiedKFold behavior.
- [ ] Add a **pooled scoring principle** in
      `aa_pred_eval.py`: run `cross_val_predict` over the splitter, then apply
      each metric once on the pooled predictions. Tag these rows distinctly in
      the `principle` column (distinct from the existing per-fold `cv` principle)
      so both are legible in `df_eval`.
- [ ] Relax `check_n_cv` when a splitter is passed so it is not capped at the
      smallest class count; keep the cap for the int path.
- [ ] Frontend validation block for the new argument (splitter must expose
      `split`); backend trusts frontend.
- [ ] numpydoc update (named `Returns`, `Examples` include) documenting the new
      argument and the pooled `principle`.

## KPIs / Acceptance criteria

- [ ] `AAPred(models=[SVC(kernel="linear")]).eval(X, y, cv=LeaveOneOut(), metrics=["balanced_accuracy"])`
      returns a score equal within `1e-9` to
      `balanced_accuracy_score(y, cross_val_predict(SVC(kernel="linear"), X, y, cv=LeaveOneOut()))`.
- [ ] Default integer-`n_cv` `df_eval` output **byte-identical** to current
      (regression-tested).
- [ ] The γ-secretase Appendix-3 `balanced_acc` helper and its call sites are
      removed; the affected figures (optimization sweep, baseline-comparison
      bars) render **byte-identical**.
- [ ] Covered by ≥1 unit test and ≥1 property test; example notebook runs under
      the nbmake CI gate.

## Scope / non-goals

- Core feature (`topic:core`); no new dependencies, no `pro` gating.
- Not adding bootstrap confidence intervals or paired ΔMCC — that is #91.
- Not adding a train-only fold-augmentation set (the PU "fair fold") — that is
  intentionally left as user code for now.

## Dependencies

- relates #91 (repeated-CV / bootstrap CIs), #335 (baseline in `AAPred.eval`).
- surfaced while reducing the γ-secretase Appendix-3 boilerplate (cf. #305).

## Standards checklist

- [ ] **CONFIRM-FIRST**: touches the public `AAPred.eval` signature (additive,
      backward compatible) — flag for review.
- [ ] Frontend/backend split honored; validation block; backend trusts frontend.
- [ ] numpydoc docstring (named `Returns`, per-method `Examples` include).
- [ ] tests (unit + property/golden); reproducibility (`random_state` contract).
- [ ] no `print()` (`ut.print_out`); bare `ValueError`/`RuntimeError`; no
      `aaanalysis._utils.*` imports outside `utils.py`.

**Comment by breimanntools** [comment]
Sibling issue: #398 (out-of-fold per-sample scoring). Both were surfaced together while shrinking the γ-secretase use-case Appendix 3; each is independently done-able and output-preserving.
